fast_pauli/cpp/include/__pauli_op.hpp

```diff
@@ -245,29 +245,40 @@ template <std::floating_point T, typename H = std::complex<T>> struct PauliOp
         coefficients.reserve(init_capacity);
         strings.reserve(init_capacity);
 
-        std::unordered_map<PauliString, size_t> dedupe_strings;
+        size_t const n_threads = omp_get_max_threads();
+        std::vector<std::unordered_map<PauliString, H>> dedupe_strings(n_threads);
+        fmt::print("n_threads: {} lhs.size: {} rhs.size: {}\n", n_threads, lhs.n_pauli_strings(),
+                   rhs.n_pauli_strings());
 
+#pragma omp parallel for schedule(static)
```

Comment:
it would be beneficial to have `collapse(2)` to maximize parallelization

Reply:
Good catch, adding it